catch_analysis_tools/astrometry.py

```diff
@@ -0,0 +1,435 @@
+    ax.legend()
+
+    return fig, ax
+
+
+def create_header(image, wcs_solution, zp, unc, source_list, matched_idx, colored_idx, input_fits):
+    """
+    Create and write a FITS file with calibrated header and source tables.
+
+    Parameters
+    ----------
+    image : array_like
+        Original 2D image data array.
+    wcs_solution : astropy.wcs.WCS
+        WCS solution for the image.
+    zp : float
+        Zero-point magnitude.
+    unc : float
+        Uncertainty of the zero-point.
+    source_list : pd.DataFrame
+        Table of detected sources.
+    matched_idx : array_like
+        Indices for matched catalog sources.
+    colored_idx : array_like
+        Indices for color-correction sources.
+    input_fits : str, optional
+        Filename for the input FITS file.
+
+    Returns
+    -------
+    None
+    """
+    image_arr = np.asarray(image)
+    primary_hdu = fits.PrimaryHDU(data=image_arr, header=wcs_solution.to_header())
+    primary_hdu.header['ZP'] = zp
+    primary_hdu.header['ZP_STD'] = unc
+    primary_hdu.header['SUV_FLT'] = 'r'
+    primary_hdu.header['REF_CATA'] = 'PANSTARR'
+    primary_hdu.header['REF_FLT'] = 'g'
+    primary_hdu.header['COR_COR'] = 'Y'
+    primary_hdu.header['CAT_COR'] = 'g-r'
+    source_list_clean = source_list.applymap(lambda x: x.filled(np.nan) if hasattr(x, 'filled') else x)
+    detected_hdu = fits.BinTableHDU(Table.from_pandas(source_list_clean), name='DETECTED_SOURCES')
+    if not source_list_clean.empty:
+        matched_hdu = fits.BinTableHDU(Table.from_pandas(source_list_clean.iloc[matched_idx].reset_index(drop=True)), name='SELECTED_STARS')
+        colored_hdu = fits.BinTableHDU(Table.from_pandas(source_list_clean.iloc[colored_idx].reset_index(drop=True)), name='START_COLOR_CORRECTION')
+    else:
+        matched_hdu = fits.BinTableHDU(name='SELECTED_STARS')
+        colored_hdu = fits.BinTableHDU(name='START_COLOR_CORRECTION')
+    hdul = fits.HDUList([primary_hdu, detected_hdu, matched_hdu, colored_hdu])
+    hdul.writeto(input_fits, overwrite=True)
+
+
+def cleanup_files(file_base):
+    """
+    Remove temporary files generated during the processing pipeline.
+
+    Parameters
+    ----------
+    file_base : str
+        Base filename (without extension) for the files to remove.
+
+    Returns
+    -------
+    None
+    """
+    extensions = ['.axy', '.corr', '.match', '.new', '.rdls', '.solved',
+                  '-ngc.png', '-objs.png', '-indx.png', '-indx.xyls']
+    for ext in extensions:
+        fname = f"{file_base}{ext}"
+        if os.path.exists(fname):
+            os.remove(fname)
+
+
+if __name__ == "__main__":
+    # Configuration
+    target_name  = "103P"
+    data_sources = "atlas_mauna_loa"
+    pixel_scale  = 1.86     # arcsec/pixel
+    snr          = 7.0      # detection threshold
+
+    input_fits, file_base, hdulist, image = load_fits_image(target_name, data_sources)
```

Comment:
Given that I recommend removing `load_fits_image`, perhaps this __main__ function should use the name of a fits image from the command line?